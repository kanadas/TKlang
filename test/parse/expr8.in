//silnia
let f = (\f n -> if n > 1 then n * f (n - 1) else n) in f f 10
