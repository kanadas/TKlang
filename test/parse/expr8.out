parse/expr8.in

Parse Successful!

[Abstract Syntax]

ELet (Ident "f") (ELambda [Ident "f",Ident "n"] (EIf (ERel (EIdent (Ident "n")) (RelOp ">") (EInt 1)) (EMul (EIdent (Ident "n")) (EApp (EIdent (Ident "f")) (ESub (EIdent (Ident "n")) (EInt 1)))) (EIdent (Ident "n")))) (EApp (EApp (EIdent (Ident "f")) (EIdent (Ident "f"))) (EInt 10))

[Linearized tree]

let f = (\ f n -> if n > 1 then n * f (n - 1) else n) in f f 10
